src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1508,13 +1511,20 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 			return NULL;
 #ifdef TARGET_ARM64
 		return emit_simd_ins_for_sig (cfg, klass, OP_XOP_OVR_X_X, INTRINS_AARCH64_ADV_SIMD_FSQRT, arg0_type, fsig, args);
-#elif TARGET_AMD64
+#elif defined(TARGET_AMD64)
 		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
 		int size = mono_class_value_size (arg_class, NULL);
-		if ( size != 16 ) 		// Only works with Vector128
+		if (size != 16) 		// Only works with Vector128
 			return NULL;
 
-		int instc0 = arg0_type == MONO_TYPE_R4 ? INTRINS_SSE_SQRT_PS : INTRINS_SSE_SQRT_PD;
+		int instc0 = INTRINS_SSE_SQRT_PD;
+		MonoCPUFeatures feature = MONO_CPU_X86_SSE2;
+		if (arg0_type == MONO_TYPE_R4) {
+			instc0 = INTRINS_SSE_SQRT_PS;
+			feature = MONO_CPU_X86_SSE;
```

Comment:
Every CPU that runs x86-64 also supports SSE2, so this refinement should not be needed. This will have an effect only on Pentium III and such.